packages/library-base/src/commonMain/kotlin/io/realm/kotlin/log/RealmLog.kt

```diff
@@ -0,0 +1,365 @@
+package io.realm.kotlin.log
+
+import io.realm.kotlin.Realm
+import io.realm.kotlin.internal.interop.CoreLogLevel
+import io.realm.kotlin.internal.interop.LogCallback
+import io.realm.kotlin.internal.interop.RealmInterop
+import io.realm.kotlin.internal.interop.SynchronizableObject
+import io.realm.kotlin.internal.platform.createDefaultSystemLogger
+import io.realm.kotlin.log.RealmLog.add
+import io.realm.kotlin.log.RealmLog.addDefaultSystemLogger
+
+/**
+ * Global logger class used by all Realm components.
+ *
+ * By default all logs will go to a default system logger that will depend on the system. See
+ * [addDefaultSystemLogger] for more details.
+ *
+ * Custom loggers can be added by registering a class implementing [RealmLogger] using [add].
+ *
+ * When logging messages, it it possible to use a subset of String format options as known from
+ * Java. Only `%s`, `%d` and `%f` are supported. See https://stackoverflow.com/a/64499248/1389357
+ * and https://youtrack.jetbrains.com/issue/KT-25506 for more information.
+ */
+public object RealmLog {
```

Comment:
I guess this should still just be an interface that is instantiated to a single instance like. 

```
public interface RealmLog {
}

public val REALM_LOG: RealmLog
```
Probably not the right suggestion, but something hiding the internals from users. But maybe I am just missing something, because I didn't expect all `trace`, `debug`, etc. methods to be exposed to the users 🤔

Reply:
All logger methods have been made internal